### ISSUE TITLE
`FieldSpec` and `resolve` does not support list access

### ISSUE THREAD

**Issue by majkrzak** [nolar/kopf#1102]
### Long story short

Currently it is not possible to specify the list fields like `status.containerStatuses[0].state` via the `field` parameter of the handlers.

### Kopf version

1.37.1

### Kubernetes version

1.27.6+k3s1

### Python version

3.11.6

**Comment by nolar** [comment]
Yes, it is impossible via the `field='…'` filter, and making it possible will add unnecessary complexity due to introducing an internal mini-DSL for such expressions. 

It is possible in general, though. The field-value-filter is just a common shortcut to a generic callable filter. It is possible to make this:

```python
import kopf

def is_of_interest(status, **_) -> bool:
  return status['containerStatuses'][0]['state'] == '…'

@kopf.on.update('pods', when=is_of_interest)
def update_fn(**_):
    …
```

This will additionally allow checking any index (not just `[0]`), any boolean combination of several containers (`all()`/`any()`), or any other sophisticated filtering logic.
